plugins/GSdx/GSRendererHW.cpp

```diff
@@ -117,13 +117,44 @@ void GSRendererHW::SetScaling()
 	{
 		if (m_height == m_custom_height)
 		{
-			float ratio = ceil(static_cast<float>(m_height) / crtc_size.y);
-			float buffer_scale_offset = (m_large_framebuffer) ? ratio : 0.5f;
-			ratio = round(ratio + buffer_scale_offset);
+			const float scaling_ratio = ceil(static_cast<float>(m_custom_height) / crtc_size.y);
+			// Avoid using a scissor value which is too high, developers can even leave the scissor to max (2047)
+			// at some cases when they don't want to limit the rendering size. Our assumption is that developers
+			// set the scissor to the actual data in the buffer. Let's use the scissoring value only at such cases
+			const int scissor_height = std::min(640u, (m_context->SCISSOR.SCAY1 - m_context->SCISSOR.SCAY0) + 1);
```

Comment:
But I guess most of your gain come from it. What happen if you keep `640` instead.